invenio_records_resources/services/records/queryparser/transformer.py

```diff
@@ -33,15 +33,55 @@ def term_name(self):
         """Get the term name."""
         return self._term_name
 
-    def map_word(self, node):
+    def map_word(self, node, **kwargs):
         """Modify a word node."""
         return self._word_fun(node) if self._word_fun else node
 
-    def map_phrase(self, node):
+    def map_phrase(self, node, **kwargs):
         """Modify a phrase node."""
         return self._phrase_fun(node) if self._phrase_fun else node
 
 
+class RestrictedTerm:
+    """Class used to apply specific permissions to search."""
+
+    def __init__(self, permission, term_name=None, word=None, phrase=None):
+        self.permission = permission
+        self._word_fun = word
+        self._phrase_fun = phrase
+        self._term_name = term_name
+
+    def allows(self, identity):
+        if not (self._word_fun or self._phrase_fun):
+            return self.permission.allows(identity)
+        else:
```

Comment:
this part a little bit flaky, I didn't have a better idea at the moment

Reply:
Honestly, this assumes that you know a lot how to call the class, the name is also confusing to me. Plus missing a nice docstring :)

While the FieldValueMapper maps a search term to a specific field, as a shortcut (we should have better described it in the class), this one restricts searches.

What about splitting RestrictedTerm in 2, with specific behaviours instead?
```
"internal_notes.timestamp": RestrictedField(administration_permission),  # does not accept other params
"_exists_": RestrictedValue( administration_permission, word=word_internal_notes),   # requires word or phrase
```
because, in the end, we restrict a field, or a value (which happens to be a field when using the `_exist_` search operator).

This should allow you to have better class implementations, without mixing them with that `allows` and the weird if inside.